workspaces/ui-v2/src/pages/docs/components/ShapeEditor.tsx

```diff
@@ -0,0 +1,464 @@
+import React, { FC, useMemo } from 'react';
```

Comment:
Thoughts on splitting this component up into multiple files? We could create a folder `ShapeEditor` with an index.ts file to match the export structure. Think this might continue to grow and a separation of files might be helpful here

Reply:
It's all tightly connected now, not yet stable and there's no reason why anything other component should use some of these internal components directly. So lets leave that for now.

Reply:
Sure, I don't think we'd intend any of these internal components directly even if we did split out these into different files. But there is probably a point where this shape editor might be easier to reason able in multiple files - since we are still iterating on this, it seems fine to keep as is